assets/queries/openAPI/empty_array/query.rego

```diff
@@ -0,0 +1,52 @@
+package Cx
+
+import data.generic.openapi as openapi_lib
+
+CxPolicy[result] {
+	doc := input.document[i]
+	openapi_lib.check_openapi(doc) != "undefined"
+
+	[path, value] := walk(doc)
+
+	property := value[x]
+	is_empty_array(property)
+
+	count(path) != 0
+
+	result := {
+		"documentId": doc.id,
+		"searchKey": sprintf("%s.%s", [openapi_lib.concat_path(path), x]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": "The array is not empty",
+		"keyActualValue": "The array is empty",
+	}
+}
+
+CxPolicy[result] {
+	doc := input.document[i]
+	openapi_lib.check_openapi(doc) != "undefined"
+
+	[path, value] := walk(doc)
+
+	property := value[x]
+	is_empty_array(property)
+
+	count(path) == 0
```

Comment:
This will ever happen? Since path includes `value` key, I think it will never be an empty array, no?

Reply:
Because of `property := value[x]`, it happens. But with your suggestion, no 🙂